crates/router/src/core/payments/operations/payment_confirm.rs

```diff
@@ -28,6 +28,52 @@ use crate::{
 #[operation(ops = "all", flow = "authorize")]
 pub struct PaymentConfirm;
 
+impl PaymentConfirm {
+    #[inline(always)]
+    fn make_new_payment_attempt(
+        request: &api::PaymentsRequest,
+        old_payment_attempt: storage::PaymentAttempt,
+    ) -> storage::PaymentAttemptNew {
+        let created_at @ modified_at @ last_synced = Some(common_utils::date_time::now());
+
+        storage::PaymentAttemptNew {
+            payment_id: old_payment_attempt.payment_id,
+            merchant_id: old_payment_attempt.merchant_id,
+            attempt_id: uuid::Uuid::new_v4().simple().to_string(),
+            status: helpers::payment_attempt_status_fsm(&request.payment_method_data, Some(true)),
+            amount: old_payment_attempt.amount,
+            currency: old_payment_attempt.currency,
+            save_to_locker: old_payment_attempt.save_to_locker,
+            connector: None,
+            error_message: None,
+            offer_amount: old_payment_attempt.offer_amount,
+            surcharge_amount: old_payment_attempt.surcharge_amount,
+            tax_amount: old_payment_attempt.tax_amount,
+            payment_method_id: None,
+            payment_method: None,
+            capture_method: old_payment_attempt.capture_method,
+            capture_on: old_payment_attempt.capture_on,
+            confirm: old_payment_attempt.confirm,
+            authentication_type: old_payment_attempt.authentication_type,
```

Comment:
use the authentication_type from the request. And add comments for fields also, why we are taking data from the old_payment_attempt and not from the request.